src/widget/wglwidgetqopengl.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#ifndef WGLWIDGET_H
+#error "Do not include this file, include wglwidget.h instead"
+#endif
+
+#include <QWidget>
+
+////////////////////////////////
+// QOpenGLWindow based WGLWidget
+////////////////////////////////
+
+class QPaintDevice;
+class OpenGLWindow;
+
+class WGLWidget : public QWidget {
+  private:
+    OpenGLWindow* m_pOpenGLWindow{};
+    QWidget* m_pContainerWidget{};
+
+  public:
+    WGLWidget(QWidget* parent);
+    ~WGLWidget();
+
+    bool isContextValid() const;
+    bool isContextSharing() const;
+
+    bool shouldRender() const;
+
+    void makeCurrentIfNeeded();
+    void doneCurrent();
+
+    void swapBuffers();
+    void resizeEvent(QResizeEvent* event) override;
+
+    void showEvent(QShowEvent* event) override;
+
+    // called (indirectly) by WaveformWidgetFactory
+    virtual void renderGL();
+    // called by OpenGLWindow
+    virtual void resizeGL(int w, int h);
+    virtual void handleEventFromWindow(QEvent* ev);
```

Comment:
The original event(QEvent* ev) is already virtual. 
This function implements an extra entry point for events. User of the base class will bypass the wrapper. 
Is this desired? Can you drop a comment why?

Reply:
If I understand you correct, you are proposing to simply call event(ev) from the OpenGLWindow. Sure, that will work. I will make the change.

Reply:
Thank you.

Reply:
Ok, this is not possible. The problem is that event(QEvent* ev) is a protected member of QWidget, so OpenGLWindow cannot call it.

Reply:
Done. Please confirm this what you meant.